script/tool/lib/src/native_test_command.dart

```diff
@@ -465,7 +465,7 @@ this command.
       _printRunningExampleTestsMessage(example, platform);
       final int exitCode = await _xcode.runXcodeBuild(
         example.directory,
-        actions: <String>['test'],
+        actions: <String>['clean', 'test'],
```

Comment:
I think we should put a code comment here explaining why the `clean` call is here, since it's non-obvious.